code/game/objects/structures/doors.dm

```diff
@@ -183,11 +183,12 @@
 	if(switching_states)
 		return FALSE
 	if(door_opened)
-		to_chat(user, span_notice("I cannot lock \the [src] while it is open."))
 		return FALSE
 	return ..()
 
 /obj/structure/door/attackby(obj/item/I, mob/user)
+	if(I.has_access())
```

Comment:
Why is this before the state check? And this doesn't fight the lock signals right